app/src/main.rs

```diff
@@ -45,43 +44,36 @@ fn main(_argc: isize, _argv: *const *const u8) -> isize {
         target_tests::run_test_suite()
     } else {
         println!("Starting game!");
-        main_game()
+        register_power_callback();
+        main_game();
+        shutdown();
+        0
     }
 }
 
 fn main_game() -> isize {
-    println!("Hello Rust!");
-    register_power_callback();
-
-    // Setup the wiimote
-    Input::init(ControllerType::Wii);
-    let wii_mote = Input::new(ControllerType::Wii, ControllerPort::One);
-    wii_mote
-        .as_wpad()
-        .set_data_format(WPadDataFormat::ButtonsAccelIR);
-
-    // Setup the ECS environment.
-    let mut world = World::new();
-    batch_spawn_entities(&mut world, 5);
-    let mut velocity_query = PreparedQuery::<&mut Velocity>::default();
-    let mut all_query = PreparedQuery::<(&mut Position, &mut Velocity)>::default();
-
-    // Kickstart main loop.
+    // register_power_callback();
+    let mut game_state = GameState::new();
+    let mut input_manager = InputManager::new();
     let renderer = Renderer::new();
+
     while KEEP_RUNNING.load(Ordering::SeqCst) {
-        Input::update(ControllerType::Wii);
-        if wii_mote.is_button_down(Button::Home) {
+        let controls = input_manager.update();
+
+        let changes = Changes {
+            controls,
+            delta_time_ms: 100,
+        };
+        let should_continue = game_state.update(&changes);
+        if !should_continue {
             break;
         }
-        if wii_mote.is_button_down(Button::One) {
-            system_shake_wii(&mut world, &mut velocity_query);
-        }
-        system_bounce_bounds(&mut world, &mut all_query);
-        system_integrate_motion(&mut world, &mut all_query);
 
-        renderer.render_world(&world);
+        renderer.render_world(&game_state.world);
     }
-    shutdown();
+    println!("Test me");
+    // shutdown();
```

Comment:
Feel free to remove the commented-out code, not only this line as well :)

Reply:
This garbage should not have been committed by me; it was unfinished testing code from another branch 😅